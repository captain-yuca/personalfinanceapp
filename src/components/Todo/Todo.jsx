import React from 'react'
import PropTypes from 'prop-types'

import { Wrapper, Title } from './Todo.styles'

import TodoList from '../TodoList'
import TodoInput from '../TodoInput'

const propTypes = {
  todos: PropTypes.array,
  addTodo: PropTypes.func,
  removeTodo: PropTypes.func,
}

function Todo({ todos, addTodo, removeTodo }) {
  return (
    <Wrapper>
      <Title>TODO APP EXAMPLE</Title>
      <TodoList todos={todos} removeTodo={removeTodo} />
      <TodoInput addTodo={addTodo} />
    </Wrapper>
  )
}

Todo.propTypes = propTypes

export default Todo
