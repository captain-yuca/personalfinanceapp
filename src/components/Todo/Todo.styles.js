import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 320px;
  margin: 0 auto;
`

export const Title = styled.h2`
  font-size: 36px;
  margin-top: 24px;
  color: ${({ theme }) => theme.colors.text};
`
