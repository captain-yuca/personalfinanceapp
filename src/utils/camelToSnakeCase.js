export const camelToSnakeCase = str => str.replace(/([A-Z])/g, group => `_${group.toLowerCase()}`)
