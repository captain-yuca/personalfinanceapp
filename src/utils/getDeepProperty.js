export const getDeepProperty = (path, obj) =>
  path.split('.').reduce((prev, curr) => (prev ? prev[curr] : null), obj || null)
