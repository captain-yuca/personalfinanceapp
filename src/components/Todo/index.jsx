import React from 'react'
import Todo from './Todo'

// hooks
import useTodos from '../../hooks/useTodos'

function Wrapper(props) {
  const todos = useTodos()
  return <Todo {...props} {...todos} />
}

export default Wrapper
