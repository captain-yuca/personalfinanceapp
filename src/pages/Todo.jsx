import React from 'react'
import Helmet from 'react-helmet'

import Todo from '../components/Todo'

export default function TodoPage() {
  return (
    <>
      <Helmet>
        <title>Todo app | React MAD Boiler</title>
      </Helmet>
      <Todo />
    </>
  )
}
