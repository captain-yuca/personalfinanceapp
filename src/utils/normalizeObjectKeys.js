import { camelToSnakeCase } from './camelToSnakeCase'
import { snakeToCamelCase } from './snakeToCamelCase'

export const normalizeObjectKeys = (obj, toSnake = false) =>
  Object.entries(obj).reduce((accum, [rawKey, value]) => {
    const key = toSnake ? camelToSnakeCase(rawKey) : snakeToCamelCase(rawKey)
    return { ...accum, [key]: value }
  }, {})
