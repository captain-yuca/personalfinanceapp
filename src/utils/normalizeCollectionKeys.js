import { normalizeObjectKeys } from './normalizeObjectKeys'

export const normalizeCollectionKeys = (collection, toSnake = false) =>
  collection.map(item => normalizeObjectKeys(item, toSnake))
