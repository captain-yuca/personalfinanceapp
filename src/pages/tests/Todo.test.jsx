import React from 'react'
import { render } from '../../test.utils'

import Todo from '../Todo'

describe('Todo page', () => {
  it('should render correctly', () => {
    const { container } = render(<Todo />)
    expect(container).toMatchSnapshot()
  })
})
