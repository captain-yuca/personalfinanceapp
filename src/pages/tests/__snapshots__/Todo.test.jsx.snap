// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo page should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 320px;
  margin: 0 auto;
}

.c1 {
  font-size: 36px;
  margin-top: 24px;
  color: #ffffff;
}

.c2 {
  margin-top: 12px;
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c3 {
  font-size: 16px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  border-radius: 8px;
  border: 2px solid #ffffff;
  background-color: #212121;
  padding: 8px;
  color: #ffffff;
  outline: none;
}

.c4 {
  margin-left: 16px;
  padding: 8px 16px;
  border: 2px solid #ffffff;
  background-color: #212121;
  border-radius: 8px;
  color: #ffffff;
  outline: none;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      TODO APP EXAMPLE
    </h2>
    <div
      class="empty"
    >
      No todos :)
    </div>
    <div
      class="c2"
    >
      <input
        class="c3"
        placeholder="Enter new todo"
        type="text"
        value=""
      />
      <button
        class="c4"
      >
        Add
      </button>
    </div>
  </div>
</div>
`;
